[PPM-292]  Remove client link measuruments 11k

VladyslavTupikin opened the pull request:
Need to remove next messages and all related code:

```
ACTION_CONTROL_CLIENT_LINK_MEASUREMENT_11K_REQUEST
ACTION_CONTROL_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE
````

Commit:
common: tlvf: Remove deprecated VS messages

Need to remove deprecated vendor-specific
messages:
ACTION_CONTROL_CLIENT_LINK_MEASUREMENT_11K_REQUEST
ACTION_CONTROL_CLIENT_LINK_MEASUREMENT_11K_RESPONSE

Also need to remove deprecated code from
autogenerated file.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove case from VS message handler

Need to remove case for processing
deprecated vendor-specific message
ACTION_CONTROL_CLIENT_LINK_MEASUREMENT_11K_REQUEST
from vs hanler.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: son_management: Remove case from handler

Need to remove case for depricated CLI message
ACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: beerocks_cli: Remove deprecated CLI command

Need to remove deprecated CLI command
client_link_measurement_11k_req
and all releated code.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: tlvf: Remove deprecated VS message

Need to remove deprecated vendor-specific
messages:
ACTION_MONITOR_CLIENT_LINK_MEASUREMENTS_11K_REQUEST
ACTION_MONITOR_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE

Also need to remove all related code
from autogenerated files.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agnet: son_slave: Remove deprecated message

Need to remove deprecated case for
ACTION_MONITOR_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE
from VS message handler.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: fronthaul_manager: Remove deprecated method

Need to remove deprecated method
debug_link_measurements_11k_response().

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Remove deprecated event

Need to remove deprecated event
RRM_Link_Measurement_Response.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Remove deprecated event from dwpal

Need to remove deprecated event
RRM_Link_Measurement_Response from dwpal.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Remove deprecated event from nl80211

Need to remove deprecated event
RRM_Link_Measurement_Response
from nl80211.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: monitor_thread: Remove deprecated event

Need to remove case for procesing
deprecated event RRM_Link_Measurement_Response.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: monitor_thread: Remove depredcated case from handler

Need to remove deprecated case for
VS message ACTION_MONITOR_CLIENT_LINK_MEASUREMENT_11K_REQUEST
from CS message handler.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: son_master: Remove case for deprecated message

Need to remove case from handler for
deprecated VS message
ACTION_CONTROL_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

adam1985d commented on agent/src/beerocks/fronthaul_manager/monitor/monitor_thread.cpp:
```diff
@@ -1760,46 +1760,6 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
 
     } break;
 
-    case Event::RRM_Link_Measurement_Response: {
```
@arnout is this message really deprecated or should it remain and its information is translated to a non-VS message?

adam1985d commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_action.yaml:
```diff
@@ -271,8 +271,7 @@ eActionOp_MONITOR:
   ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE: 34
-  ACTION_MONITOR_CLIENT_LINK_MEASUREMENT_11K_REQUEST: 35
-  ACTION_MONITOR_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE: 36
+
```
please remove the line space

VladyslavTupikin replied:
fixed

adam1985d replied:
the line space is still here
(nitpick)

adam1985d commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_monitor.yaml:
```diff
@@ -159,14 +159,6 @@ cACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE:
   _type: class
   params: sStatisticsResponse11k
 
```
this commit and the previous one (of the auto-generated code) should be in the same commit IMO

VladyslavTupikin replied:
accepted

adam1985d commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_action.yaml:
```diff
@@ -330,8 +330,8 @@ eActionOp_CLI:
   ACTION_CLI_CLIENT_BSS_STEER_REQUEST: 83
-  ACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST: 84
 
+  ACTION_CLI_CLIENT_CHANNEL_LOAD_11K_REQUEST: 85
```
this commit message is about removal - not addition of new CMDUs

VladyslavTupikin replied:
fixed

adam1985d commented on common/beerocks/tlvf/AutoGenerated/include/beerocks/tlvf/beerocks_message_cli.h:
```diff
@@ -458,18 +458,19 @@ class cACTION_CLI_CLIENT_BSS_STEER_REQUEST : public BaseClass
-        cACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST(uint8_t* buff, size_t buff_len, bool parse = false);
-        explicit cACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST(std::shared_ptr<BaseClass> base, bool parse = false);
-        ~cACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST();
+        cACTION_CLI_CLIENT_CHANNEL_LOAD_11K_REQUEST(uint8_t* buff, size_t buff_len, bool parse = false);
```
again - please provide the tlvf and the autogenerated files in the same commit - no added value in splitting them up

VladyslavTupikin replied:
fixed

adam1985d commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_action.yaml:
```diff
@@ -87,8 +87,7 @@ eActionOp_CONTROL:
   ACTION_CONTROL_CLIENT_STATISTICS_11K_REQUEST: 121
   ACTION_CONTROL_CLIENT_STATISTICS_11K_RESPONSE: 122
-  ACTION_CONTROL_CLIENT_LINK_MEASUREMENT_11K_REQUEST: 123
-  ACTION_CONTROL_CLIENT_LINK_MEASUREMENTS_11K_RESPONSE: 124
```
please provide the tlvf changes and autogenerated files in the same commit.

and please remove the added line space

VladyslavTupikin replied:
fixed

adam1985d reviewed (changes requested):
the only reason I am not approving is that one of the commits adds a CMDU (either by mistake or by purpose) but the commit message says nothing about it.
General suggestions:
1. replace in all commit messages the `Need to remove the..." with "Removed the.."
2. tlvf changes and autogenerated files should be on the same commit IMO
all are nitpicks.

the only real issue: fix the commit that adds new CMDU and trigger re-review

Commit:
common: tlvf: Remove deprecated message

Need to merove deprecated message
ACTION_CLI_CLIENT_LINK_MEASUREMENT_11K_REQUEST
and all related code from autogenerated files.

https://jira.prplfoundation.org/browse/PPM-292

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

mariomaz reviewed (approved):
LGTM

adam1985d reviewed (approved):
I added (or actually unresolved) one comment, but it is a nitpick so approving
